src/Xrefcheck/Verify.hs

```diff
@@ -160,9 +190,13 @@ verifyRepo
 
   progressRef <- newIORef $ initVerifyProgress (map snd toScan)
 
-  withAsync (printer progressRef) $ \_ ->
-    fmap fold . forConcurrently toScan $ \(file, ref) ->
-      verifyReference config mode progressRef repoInfo' root file ref
+  accumulated <- withAsync (printer progressRef) $ \_ ->
+    forConcurrentlyCaching toScan (\(_, Reference{..}) -> case locationType rLink of
+      ExternalLoc -> CacheUnderKey rLink
+      _           -> NoCaching) (\(file, ref) -> do
+      VerifyResult results <- verifyReference config mode progressRef repoInfo' root file ref
+      return $ Left results)
```

Comment:
Looks like here you wrap into `Left` (and never into `Right`), and later evaluate `lefts` which brings us back.

At this point `Either`s are probably gone, no need to work with them.